common/src/main/java/us/ilite/common/lib/control/PIDController.java

```diff
@@ -0,0 +1,337 @@
+package us.ilite.common.lib.control;
+
+import com.flybotix.hfr.util.log.ELevel;
+import com.flybotix.hfr.util.log.ILog;
+import com.flybotix.hfr.util.log.Logger;
+
+import com.team254.lib.util.Util;
+/**
+ * This class implements a PID Control Loop.
+ * 
+ * Does all computation synchronously (i.e. the calculate() function must be called by the user from his own thread)
+ */
+public class PIDController {
+
+    private ILog mLogger = Logger.createLog(this.getClass());
+
+    private double m_P; // factor for "proportional" control
+    private double m_I; // factor for "integral" control
+    private double m_D; // factor for "derivative" control
+    private double m_F; // factor for feed forward gain
+    private double m_maximumOutput = 10.0; // |maximum output|
+    private double m_minimumOutput = -10.0; // |minimum output|
+    private double m_maximumInput = 0.0; // maximum input - limit setpoint to
+                                         // this
+    private double m_minimumInput = 0.0; // minimum input - limit setpoint to
+                                         // this
+    private boolean m_continuous = false; // do the endpoints wrap around? eg.
+                                          // Absolute encoder
+    private double m_prevError = 0.0; // the prior sensor input (used to compute
+                                      // velocity)
+    private double m_totalError = 0.0; // the sum of the errors for use in the
+                                       // integral calc
+    private double m_setpoint = 0.0;
+    private double m_error = 100.0;
+    private double m_result = 0.0;
+    private double m_last_input = Double.NaN;
```

Comment:
Do we really want the last state to be NaN initially? The problem with NaN is that every calculation that is performed on NaN will return NaN. Also what will happen if NaN is sent to dashboard?